llvm/lib/CodeGen/SelectionDAG/SelectionDAG.cpp

```diff
@@ -5091,8 +5101,18 @@ bool SelectionDAG::canCreateUndefOrPoison(SDValue Op, const APInt &DemandedElts,
   case ISD::BITCAST:
   case ISD::BUILD_VECTOR:
   case ISD::BUILD_PAIR:
+  case ISD::SELECT:
```

Comment:
Do we need to worry about Fast math flags on SELECT?

Reply:
Thanks! After several years I still did not know that `select` could have flags.
I really think that we shuold document such things also for ISD opcodes (at least in ISDOpcodes.h).

I'll try to add an assert here to make sure that none of the instructions in the group have flags. And then I'll move the handling of SELECT further down in this switch, to make sure that we consider the flags in some way.